docs/ways-of-working.md

```diff
@@ -0,0 +1,81 @@
+
+```
+### Migration Steps
+
+[Description of what needs to be done to adopt the code change with examples]
+```
+
+### Workflow
+* Assign yourself to the issue
+* Create a new branch off **develop** using pattern: `gh-[issue number]-[issue-title]`
+* Commit your changes prefixing your descriptive commit title, like so: `gh-[issue-number]: [commit title]`
+* Check and push your changes
+* Create a pull request (PR) to merge your branch into **develop**
+* If you named the branch and PR correctly, the PR should have "Resolve #[issue-number]" automatically added to the description after it is made. If it doesn't, then please add the issue it will resolve as a "Linked issue"
+* If there is a significant change, please follow the same process to document the change in [gaffer-doc](https://github.com/gchq/gaffer-doc)
+* The pull request will be reviewed and following any changes and approval your branch will be squashed and merged into **develop**
+* Delete the branch
+* The issue will be closed automatically
+
+## Pull Requests
+Pull requests will undergo a review by a Gaffer committer to check the code changes are compliant with our coding style. This is a community so please be respectful of other members - offer encouragement, support and suggestions. 
+
+As described in our git branching model - please raise pull requests to merge you changes in our **develop** branch.
+
+When pull requests are accepted, the reviewer should squash and merge them. This is because it keeps the **develop** branch clean and populated with only merge commits, rather than intermediate ones. As well as this, it makes everyone's job reviewing pull requests easier as any insecure and unreviewed intermediate commits are not included into the **develop** branch.
+
+Please agree to the [GCHQ OSS Contributor License Agreement](https://github.com/GovernmentCommunicationsHeadquarters/Gaffer/wiki/GCHQ-OSS-Contributor-License-Agreement-V1.0) before submitting a pull request. Signing the CLA is enforced by the cla-assistant.
+
+## Documentation
+As mentioned before, any significant changes in a PR should be accompanied with an addition to Gaffer's documentation: [gaffer-doc](https://github.com/gchq/gaffer-doc).
+Smaller changes should be self documented in the tests. With this approach, any large feature or change has user friendly documentation, whereas technical or implementation details are documented for developers by the tests.
+
+## Coding style
+Please ensure your coding style is consistent with rest of the Gaffer project and the [Google Java Style Guide](https://google.github.io/styleguide/javaguide.html). Your changes should pass the checkstyle and spotless plugin checks when running `mvn install`, these check for code formatting and licenses and are part of the continuous integration pipeline.
```

Comment:
The spotless checks don't run as part of `mvn install` (to prevent problems with projects inheriting the POM). They need to be run on their own using `mvn spotless:check`, this is done as a separate step in the CI.